make stormpath filter conditional on missing bean,  expos…

iancorcoran opened the pull request:
Fixes #964 

…ed some filter dependencies as beans

mrioan commented on extensions/spring/stormpath-spring-webmvc/src/main/java/com/stormpath/spring/config/AbstractStormpathWebMvcConfiguration.java:
```diff
@@ -836,6 +837,7 @@ public RefreshTokenResultFactory stormpathRefreshTokenResultFactory() {
         return new DefaultRefreshTokenResultFactory(application);
     }
 
+    @Bean
```
@iancorcoran this is an abstract class. We do not define beans here. All our configurations in the web layer follow the same pattern. We have an abstract class where we implement most of the functionality and 2 concrete classes: one for Spring (in this case `StormpathWebMvcConfiguration.java`) and one for Spring Boot (StormpathWebMvcAutoConfiguration.java). The Beans are defined in these 2 clases, not in the abstract one.

mrioan commented on extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/java/com/stormpath/spring/boot/autoconfigure/StormpathWebMvcAutoConfiguration.java:
```diff
@@ -654,6 +654,7 @@ public FilterChainManager stormpathFilterChainManager() {
     }
 
     @Bean
+    @ConditionalOnMissingBean
```
Please make it `@ConditionalOnMissingbean(name = "stormpathServletContextListener")`. The lack of this explicit naming causes some issues when developers use a different version of Spring.
Additionally, if it is not much trouble, could you please add the name to some of the pre-existent beans that are missing the name please? For example, `stormpathAccessTokenResultFactory`, `stormpathFieldValueResolver`, etc do not have the name explicitly defined (there are quite a few more). Thanks in advance!!

mrioan commented:
@iancorcoran back to you. Thanks

iancorcoran commented:
@mrioan I moved bean declarations out of AbstactStormpathWebMvcConfiguration to the subclasses, and also added names to all the other ConditionalOnMissingBean annotations. Let me know if you have any further comments. Thanks!

mrioan commented:
Looks good do me. Feel free to merge when Travis passes. Thanks!. Back to you @iancorcoran
